File: Java/Problem_031.java
public class Main {
    public static void main(String[] args) {
        int target = 200;
        int[] coins = {1,2,5,10,20,50,100,200};
        long[] dp = new long[target + 1];
        dp[0] = 1;

        for (int coin : coins) {
            for (int x = coin; x <= target; x++) {
                dp[x] += dp[x - coin];
            }
        }

        System.out.println(dp[target]); // 73682
    }
}

File: Python/Problem_031.py
def solve(target: int = 200) -> int:
    coins = [1, 2, 5, 10, 20, 50, 100, 200]
    dp = [0] * (target + 1)
    dp[0] = 1

    for coin in coins:
        for x in range(coin, target + 1):
            dp[x] += dp[x - coin]

    return dp[target]

if __name__ == "__main__":
    print(solve())  # 73682
